packages/core/src/common/message-rpc/channel.ts

```diff
@@ -68,15 +69,11 @@ export interface ChannelCloseEvent {
 export type MessageProvider = () => ReadBuffer;
 
 /**
- * Helper class to implement the single channels on a {@link ChannelMultiplexer}. Simply forwards write requests to
- * the given write buffer source i.e. the main channel of the {@link ChannelMultiplexer}.
+ *  Reusable abstract {@link Channel} implementation that sets up
+ *  the basic channel event listeners and offers a generic close method.
  */
-export class ForwardingChannel implements Channel {
-
-    protected toDispose = new DisposableCollection();
-    constructor(readonly id: string, protected readonly closeHandler: () => void, protected readonly writeBufferSource: () => WriteBuffer) {
-        this.toDispose.pushAll([this.onCloseEmitter, this.onErrorEmitter, this.onMessageEmitter]);
-    }
+@injectable()
+export abstract class AbstractChannel implements Channel {
```

Comment:
Currently all channel instances are directly constructed without DI. Most of the channel classes use constructor arguments which are typically not injected so for those subclasses injection doesn't really make sense IMO (and if they should be bound in a DI container one would probably use factories).
However, adopters still might want to inject a custom channel implemention.
By adding the `@injectable` decorator to the `AbstractChannel` class we make sure that it can be reused  for this usecase.

Reply:
Alright, I'm fine with that.